xfce-extra/xfce4-kbdleds-plugin/xfce4-kbdleds-plugin-0.2.3.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit xdg-utils
+
+DESCRIPTION="A panel plug-in to show state of Caps, Num and Scroll Lock keys"
+HOMEPAGE="https://github.com/oco2000/xfce4-kbdleds-plugin"
+SRC_URI="https://github.com/oco2000/${PN}/archive/${P}.tar.gz"
+S="${WORKDIR}/${PN}-${P}"
+PATCHES=( "${FILESDIR}/fix-build-0.2.3.patch" )
+
+LICENSE="GPL-2 LGPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+RDEPEND="xfce-base/libxfce4ui
+	xfce-base/libxfce4util
+	xfce-base/xfce4-panel"
+DEPEND="${RDEPEND}
+	dev-util/intltool
+	sys-devel/gettext
+	virtual/pkgconfig"
+
+src_configure() {
+	./autogen.sh || die
```

Comment:
Nononono. There's autotools.eclass for that, and I'm pretty sure you're going to need a dep on xfce4-dev-tools.

Also, could you ask upstream to make `make distcheck` or make a ready dist tarball yourself? It'd be better than requiring all the users to do regen.

Reply:
`./autogen.sh` is what the upstream README told me to do, that’s why I did it. I tried calling `eautoreconf` in `src_prepare` (I think that’s where it’s supposed to be used?) but just got six copies of `/bin/grep: configure.??: No such file or directory` instead, followed by bailing out. The `autogen.sh` shipped by upstream runs `/usr/bin/xdt-autogen`, which is part of `xfce4-dev-tools` (yes I added the dep, thanks for noticing that).

I don’t really know much about autotools. Can you give any guidance about what I should be doing here?

Reply:
Just run `make distcheck` to make a good tarball with pregenerated files and use that instead.

Reply:
OK, where should I put the tarball though? It’s not my project, and since I’m not a Gentoo dev I can’t upload it to a Gentoo-infra-hosted home directory.

Reply:
Try `https://dev.gentoo.org/~mgorny/dist/${P}.tar.bz2` -- and diff the unpacked directory against your archive to be sure.

Reply:
Yep, that worked (and once I upgraded to unstable autoconf and automake I was able to generate a tarball with identical contents, albeit not byte-for-byte identical, I guess maybe it had different file timestamps). At this point I’m not sure whether `xfce4-dev-tools` is still needed (now that `autogen` no longer needs to run); any thoughts on that? The previous version didn’t depend on it.

Reply:
It compiled fine without it installed, so I’m guessing not (and have removed it accordingly). If you disagree, let me know and I’ll re-add it.

Reply:
@mgorny AFAIK that was all the outstanding items.